### PR TITLE
Catch xarray warning about out of range dates

### LINKED ISSUE #221: Suppress xarray warning about decoding CF
xarray emits a warning [here](https://github.com/pydata/xarray/blob/master/xarray/conventions.py#L909-L911) when decoding CF variables.  When submitting many calculations in parallel, the result is this repeated many times and making the logging harder to follow.  E.g. this snippet from a recent submission:

```
INFO:root:Getting input data: Var instance "swup_sfc" (Fri Oct 20 16:46:35 2017)
INFO:root:Getting input data: Var instance "precip" (Fri Oct 20 16:46:35 2017)
/home/s1h/anaconda/envs/py36/lib/python3.6/site-packages/xarray/conventions.py:838: UserWarning: CF decoding is overwriting dtype
  warnings.warn("CF decoding is overwriting dtype")
/home/s1h/anaconda/envs/py36/lib/python3.6/site-packages/xarray/conventions.py:838: UserWarning: CF decoding is overwriting dtype
  warnings.warn("CF decoding is overwriting dtype")
/home/s1h/anaconda/envs/py36/lib/python3.6/site-packages/xarray/conventions.py:838: UserWarning: CF decoding is overwriting dtype
  warnings.warn("CF decoding is overwriting dtype")
/home/s1h/anaconda/envs/py36/lib/python3.6/site-packages/xarray/conventions.py:838: UserWarning: CF decoding is overwriting dtype
  warnings.warn("CF decoding is overwriting dtype")
/home/s1h/anaconda/envs/py36/lib/python3.6/site-packages/xarray/conventions.py:838: UserWarning: CF decoding is overwriting dtype
  warnings.warn("CF decoding is overwriting dtype")
INFO:root:Getting input data: Var instance "swdn_sfc" (Fri Oct 20 16:46:37 2017)
INFO:root:Computing timeseries for 1983-01-01 00:00:00 -- 2012-12-31 00:00:00.
```

This decoding conversion is so central to our workflow that I don't think we should let this warning propagate.  Instead, I think we should catch it and reroute it to our log at the `logging.DEBUG` level.  This prevents it from being seen unless the user deliberately specifies such a verbose logging level.

I suspect this warning is also troublesome for new users, who might interpret as meaning that something bad has happened.

### PR DESCRIPTION
**Pull request by spencerahill**
Closes #221.

Uses [warnings.catch_warnings](https://docs.python.org/3.1/library/warnings.html) from the standard library.

Still needs tests

### CODE REVIEW

**Comment by spencerahill** [comment]
@spencerkclark this is ready as far as I can tell.  A review would be great.  @micahkim23 feel free to add comments also.  Thanks!

File: aospy/test/test_data_loader.py [hunk 1]
```diff
@@ -469,9 +475,17 @@ def test_load_variable(self):
...
+        expected = _open_ds_catch_warnings(filepath)['condensation_rain']
         np.testing.assert_array_equal(result.values, expected.values)
 
+    def test_load_variable_does_not_warn(self):
+        with warnings.catch_warnings(record=True) as warnlog:
+            self.data_loader.load_variable(condensation_rain,
+                                           datetime(5, 1, 1),
+                                           datetime(5, 12, 31),
+                                           intvl_in='monthly')
+        assert len(warnlog) == 0
```
Comment (spencerkclark) [inline aospy/test/test_data_loader.py:487]: It looks like we'll need to address the `FutureWarning` in the latest version of xarray for this test to pass:
```
  /home/travis/build/spencerahill/aospy/aospy/utils/times.py:475: FutureWarning: xarray.DataArray.__contains__ currently checks membership in DataArray.coords, but in xarray v0.11 will change to check membership in array values.
    if RAW_START_DATE_STR in da:
```
I think we'll just need to change: https://github.com/spencerahill/aospy/blob/82af0ba36771bb4fd183e2d7460c278dcc45d9bb/aospy/utils/times.py#L473
to:
```python
if RAW_START_DATE_STR in da.coords:
```

File: aospy/test/test_utils_times.py [hunk 2]
```diff
@@ -160,87 +161,80 @@ def test_numpy_datetime_range_workaround():
...
-        ds[TIME_STR].attrs['calendar'] = 'noleap'
-        actual, min_yr, max_yr = numpy_datetime_workaround_encode_cf(ds)
-
-        time_desired = xr.DataArray([days, days + three_yrs],
-                                    dims=[TIME_STR])
-        desired = xr.Dataset(coords={TIME_STR: time_desired})
-        desired[TIME_STR].attrs['units'] = expected_units
-        desired[TIME_STR].attrs['calendar'] = 'noleap'
-        return actual, min_yr, max_yr, desired
+def _create_test_data(days, ref_units, expected_units):
```
Comment (spencerkclark) [inline aospy/test/test_utils_times.py]: Thanks for cleaning up these tests!  Now that this function is no longer a sub-function of `test_numpy_datetime_workaround_encode_cf`, maybe we should name it a little more specifically?  Perhaps something like: `_create_datetime_workaround_test_data`?

**Review by spencerkclark — COMMENTED** [review]
Just a couple minor comments; otherwise this looks great to me.  It will be really nice not to have to see these when running the main script or looking at CI logs.

**Comment by spencerahill** [comment]
Thanks @spencerkclark , I addressed both of those.

While I was at it, I noticed warnings from `test_tutorial`, which ultimately led me to discover that runipy [has been deprecated](https://github.com/paulgb/runipy/blob/40f2205f011b11bcf1cf9929cd1a443aaeb0e4ae/README.rst), and instead you can use Jupyter directly to [execute notebooks from within Python](https://nbconvert.readthedocs.io/en/latest/execute_api.html).  So I took the liberty of updating those as well.

I'm still getting a weird RuntimeWarning about a "sticky bit" from the tutorial, but frankly I can live with that one.  For reference, here's an example of that message:

```
========================================================================= warnings summary =========================================================================
aospy/test/test_tutorial.py::test_tutorial_notebook
  /Users/shill/Dropbox/miniconda3/envs/py36/lib/python3.6/site-packages/jupyter_client/connect.py:157: RuntimeWarning: Failed to set sticky bit on '/var/folders/_s/qf0zz57x5xzgbgn4dxbg7k_c0000gn/T': [Errno 1] Operation not permitted: '/var/folders/_s/qf0zz57x5xzgbgn4dxbg7k_c0000gn/T'
    RuntimeWarning,
```

Will merge tomorrow (Monday) unless you have anything further.  Thanks!

**Comment by spencerahill** [comment]
Woops, tutorial test is getting skipped: https://ci.appveyor.com/project/spencerahill/aospy/build/1.0.423/job/egrh2h3cgm2kanh8#L330

I probably messed up the `importorskip` calls.  Will fix tomorrow (Monday).

**Comment by spencerkclark** [comment]
> While I was at it, I noticed warnings from `test_tutorial`, which ultimately led me to discover that runipy [has been deprecated](https://github.com/paulgb/runipy/blob/40f2205f011b11bcf1cf9929cd1a443aaeb0e4ae/README.rst), and instead you can use Jupyter directly to [execute notebooks from within Python](https://nbconvert.readthedocs.io/en/latest/execute_api.html).  So I took the liberty of updating those as well.

Very nice!  

> Woops, tutorial test is getting skipped

Could the skipping just be a sign that the `importorskip` is working properly?  In other words do we just need to add `jupyter` to our build environments?

**Comment by spencerahill** [comment]
The AppVeyor failures are #238 ; everything else green.  So I'm going ahead and merging.

Thanks @spencerkclark for the review!

**Pull request merged**